hibernate-core/src/main/java/org/hibernate/LockMode.java

```diff
@@ -34,6 +34,23 @@
  * which any pessimistic lock is already held has no effect,
  * and does not force a version increment.
  * <p>
+ * When an entity is read from the database, its lock mode
+ * determines whether lost updates and non-repeatable reads
+ * are possible.
+ * <ul>
+ * <li>{@link #NONE} and {@link #READ} prevent lost updates
+ *     only for versioned entities, but do not prevent
+ *     non-repeatable reads, and do not prevent lost updates
+ *     for entities with no version attribute.
+ * <li>{@link #OPTIMISTIC} and {@link #OPTIMISTIC_FORCE_INCREMENT}
```

Comment:
> {@link #OPTIMISTIC} and {@link #OPTIMISTIC_FORCE_INCREMENT} prevent ... non-repeatable reads ...

So in strict sense, I'm not sure this is true and still depends on the isolation.  E.g., assuming RC isolation I think this results in a non-repeatable read -

```java
var b = session.find( Book.class, 1, LockModeType.OPTIMISTIC );
// updated in another Session
session.refresh( b );
```

Granted, this is "cheating" a bit, but...

Reply:
> depends on the isolation

Yeah, OK, should clarify that the assumption is RC at least.

> Granted, this is "cheating" a bit

Well yeah, I mean, if you call `refresh()` you're explicitly asking for a non-repeatable read, so I wouldn't say that counts.